### ISSUE TITLE
Docker containers not showing after update

### ISSUE THREAD

**Issue by johngo54** [nicolargo/glances#2031]
Not sure if this is the correct way; this is my first issue report.

I've been running Glances in a Docker container on three different machines: a Virtual Private Server at Strato, a Raspberry Pi 4B, and a Synology NAS. In all three cases Watchtower performed an automatic update on 10 April 2022. Afterwards, also in all three cases, when accessing Glances in a browser the other Docker containers no longer show up (as they did before). Since I'm fairly new to all this I'm quite certain that I'm missing somthing simple and obvious, perhaps.

Checking the log, it shows this message:

Error while initializing the docker plugin (No module named 'dateutil')

Thanks! 
Happy Glances user

**Comment by Wesleyl89** [comment]
Had the same error and my Home Assistant entities didn't update. I reverted back to the tag nicolargo/glances:3.2.4.2 and it works again.

**Comment by thtwice** [comment]
> Had the same error and my Home Assistant entities didn't update. I reverted back to the tag nicolargo/glances:3.2.4.2 and it works again.

Same. Can confirm that latest is broken but 3.2.4.2 works perfectly.

**Comment by RazCrimson** [comment]
I presume the issue is in the packaging of the docker image for the tag `3.2.5`

**Comment by giustanca** [comment]
Same problem here with Home Assistant after Glances update. The entities of glances in HA are not available. Reverted back to version 3.2 and everything started working again

**Comment by GuyKh** [comment]
Same here.

**Comment by nicolargo** [comment]
Hi all, sorry for the delay but i have been busy by my "real" work pasts weeks.

I just make a test on my Ubuntu system and the Docker container version 3.2.5 works on my side:

    docker pull nicolargo/glances:3.2.5-full
    docker run -d --restart="always" -p 61208-61209:61208-61209 -e GLANCES_OPT="-w" -v /var/run/docker.sock:/var/run/docker.sock:ro --pid host nicolargo/glances:3.2.5-full

![image](https://user-images.githubusercontent.com/776747/163665383-b6ab46b3-ce4b-450e-a798-52a8d8b51e7b.png)

What image name do you use ?

**Comment by phbasler** [comment]
@nicolargo I have the same problem as the others. 
My docker compose setup on a raspberry PI 4 with raspbery PI OS (Bullseye) is 

  monitoring:
    container_name: glances
    image: nicolargo/glances
    restart: unless-stopped
    pid: host
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    environment:
      - "GLANCES_OPT=--webserver --percpu"
    ports:
      - "61208:61208"

The webview still works without a problem, but the HomeAssistant Integration reports that all sensors are unavailable. If I change to image: nicolargo/glances:3.2.4.2 then the integration works again without changing anything in HomeAssistant.

**Comment by nicolargo** [comment]
@phbasler strange...

Can you run your HomeAssistant and try the following command line:

    curl -I http://localhost:61208/api/3/status
    curl http://localhost:61208/api/3/cpu
    curl http://localhost:61208/api/3/mem

**Comment by phbasler** [comment]
Sure, 

> curl -I http://localhost:61208/api/3/status

yields

> HTTP/1.0 200 OK
> Date: Sat, 16 Apr 2022 07:30:45 GMT
> Server: WSGIServer/0.2 CPython/3.9.12
> Access-Control-Allow-Origin: *
> Access-Control-Allow-Methods: GET, POST, PUT, OPTIONS
> Access-Control-Allow-Headers: Origin, Accept, Content-Type, X-Requested-With, X-CSRF-Token
> Content-Encoding: identity
> Content-Length: 0
> Content-Type: text/html; charset=UTF-8

> curl http://localhost:61208/api/3/cpu
yields
> {"total": 34.2, "user": 21.2, "nice": 0.0, "system": 12.9, "idle": 65.4, "iowait": 0.4, "irq": 0.0, "softirq": 0.1, "steal": 0.0, "guest": 0.0, "guest_nice": 0.0, "time_since_update": 55.669321060180664, "cpucore": 4, "ctx_switches": 159112, "interrupts": 94278, "soft_interrupts": 39222, "syscalls": 0}% 

and 
> curl http://localhost:61208/api/3/mem
yields
> {"total": 8281374720, "available": 6142464000, "percent": 25.8, "used": 2138910720, "free": 6142464000, "active": 905568256, "inactive": 3293409280, "buffers": 47054848, "cached": 2702909440, "shared": 527007744}%  
`

**Comment by phbasler** [comment]
@nicolargo Out of curiosity, I tried the same with my ubuntu 20.04 laptop. There I have the same results. With the latest docker build I get no sensors in HA, but with 3.2.4.2 everything works as expected.

**Comment by phbasler** [comment]
@nicolargo Did some last testing: It seems to be some dependencies. If I use the image 'image: nicolargo/glances:latest-full' then everything works as expected.

**Comment by johngo54** [comment]
Yep, all is fine now. Thank you!

**Comment by ranrinc** [comment]
Sorry for not seeing this thread, however I'm still stuck with multiple glances unable to receive any information from my Home-Assistant. Both system running non docker glances version install on Debian 11. One is actually the Proxmox OS which being monitor from the home-assistant. Any idea on how to fixed this? rather than have to run Glances on Docker?

**Comment by ranrinc** [comment]
I got this error messages and unsure what to do since I did not used glances with docker and still trying to downgrade without success. 

```
root@pve:~# glances --module-list
Error while initializing the docker plugin (No module named 'dateutil')
Plugins list: alert, amps, cloud, connections, core, cpu, diskio, folders, fs, gpu, help, ip, irq, load, mem, memswap, network, now, percpu, ports, processcount, processlist, psutilversion, quicklook, raid, sensors, smart, system, uptime, wifi
Exporters list: cassandra, couchdb, csv, elasticsearch, graph, graphite, influxdb, influxdb2, json, kafka, mqtt, opentsdb, prometheus, rabbitmq, restful, riemann, statsd, zeromq
```

**Comment by ranrinc** [comment]
> I presume the issue is in the packaging of the docker image for the tag `3.2.5`

not only for docker image and I got the same issue when running using pip3

**Comment by kelvinmo** [comment]
@ranrinc I had the same issue but was able to fix it by installing the `dateutil` module with `pip3 install python-dateutil`

**Comment by ranrinc** [comment]
> pip3 install python-dateutil

Thanks...

**Comment by Write** [comment]
Same issue, thanks to @kelvinmo

**Comment by nicolargo** [comment]
The last version (Glances docker **3.2.6.1-full**) solves this issue.

**Comment by vajonam** [comment]
on this system that works.

```
/glances # glances --module-list
Plugins list: alert, amps, cloud, connections, core, cpu, diskio, docker, folders, fs, gpu, help, ip, irq, load, mem, memswap, network, now, percpu, ports, processcount, processlist, psutilversion, quicklook, raid, sensors, smart, system, uptime, wifi
Exporters list: cassandra, couchdb, csv, elasticsearch, graph, graphite, influxdb, influxdb2, json, kafka, mqtt, opentsdb, prometheus, rabbitmq, restful, riemann, statsd, zeromq

```
the host that doesn't show the docker entities. 
```
root@hal:/app# glances --module-list
Plugins list: alert, amps, cloud, connections, containers, core, cpu, diskio, folders, fs, gpu, help, ip, irq, load, mem, memswap, network, now, percpu, ports, processcount, processlist, psutilversion, quicklook, raid, sensors, smart, system, uptime, wifi
Exporters list: cassandra, couchdb, csv, elasticsearch, graph, graphite, influxdb, influxdb2, json, kafka, mongodb, mqtt, opentsdb, prometheus, rabbitmq, restful, riemann, statsd, zeromq

```

missing the docker plugin.. ? not sure why that is.

**Comment by vajonam** [comment]
Got it. still doesn't seem to explain why the entities from one host (running NVIDIA gpu docker runtime )  are missing the docker entities.

**Comment by mvicomoya** [comment]
@nicolargo any relatively recent changes that could have broken this again?

I no longer get any data for any of my containers with latest or 3.4.0.3-full. However, downgrading to 3.2.6.1-full makes everything work again.

**Comment by nicolargo** [comment]
Nope, as said previously in the thread the **docker** plugin has been renamed to **containers**.
So the Home Assistant plugin (somewhere https://github.com/home-assistant/core/blob/dev/homeassistant/components/glances/sensor.py#L202-L226) and the Python-Glances-API client (https://github.com/search?q=repo%3Ahome-assistant-ecosystem%2Fpython-glances-api%20docker&type=code) should be adapted for Glances 3.4 and higher.

**Comment by mvicomoya** [comment]
@nicolargo, I should have been clearer in my comment. I am seeing the issue with all my containers and I'm using straight docker compose for Glances, not Home Assistant.

My Glances config file is correct with a `[containers]` section. Still, no container data with recent versions, only after downgrading I get everything to work again.

I will file a separate issue separate from this one.

**Comment by vajonam** [comment]
This actually related to the home assistant / glances api being broken. 

https://github.com/home-assistant-ecosystem/python-glances-api/issues/18#issuecomment-1791581598

Its been a long running issue no movement on that owner of the API.

**Comment by vajonam** [comment]
you can run this https://github.com/home-assistant-ecosystem/python-glances-api/issues/18#issuecomment-1758003786 inside the HA container to fix it until you update that container.
